prow/kube/prowjob.go

```diff
@@ -124,11 +116,51 @@ type ProwJobSpec struct {
 	// a Kubernetes agent
 	PodSpec *v1.PodSpec `json:"pod_spec,omitempty"`
 
+	// DecorationConfig holds configuration options for
+	// decorating PodSpecs that users provide
+	DecorationConfig *DecorationConfig `json:"decoration_config,omitempty"`
+
 	// RunAfterSuccess are jobs that should be triggered if
 	// this job runs and does not fail
 	RunAfterSuccess []ProwJobSpec `json:"run_after_success,omitempty"`
 }
 
+type DecorationConfig struct {
+	// Decorate determines if the user-provided PodSpec
+	// will be decorated with more containers to add in
+	// cloning and logging functionality.
+	Decorate bool `json:"decorate,omitempty"`
```

Comment:
Is this needed anymore now with the new struct?

Reply:
True, could check for non-nil